Modules/Loadable/Segmentations/Widgets/qMRMLSegmentEditorWidget.cxx

```diff
@@ -3118,24 +3119,28 @@ void qMRMLSegmentEditorWidget::onSelectSegmentShortcut()
   std::vector<std::string> visibleSegmentIDs;
   displayNode->GetVisibleSegmentIDs(visibleSegmentIDs);
   QString currentSegmentID = this->currentSegmentID();
-  QSet<QString> visibleSegmentIDsSet;
-  for (std::string visibleSegmentID : visibleSegmentIDs)
+
+  QModelIndexList selectedRows = d->SegmentsTableView->tableWidget()->selectionModel()->selectedRows();
+  if (selectedRows.size() == 0)
     {
-    visibleSegmentIDsSet << QString::fromStdString(visibleSegmentID);
+    return;
     }
-  QSet<QString> displayedSegmentIDsSet = d->SegmentsTableView->displayedSegmentIDs().toSet();
-  QStringList iterableSegmentIds = visibleSegmentIDsSet.intersect(displayedSegmentIDsSet).toList();
-  for (int segmentIndex = 0; segmentIndex < iterableSegmentIds.size(); segmentIndex++)
+
+  int row = selectedRows[0].row();
+  int rowCount = d->SegmentsTableView->tableWidget()->model()->rowCount();
+  int newRow = selectedRows[0].row();
+  QString newSegmentID = currentSegmentID;
+  for (int i = 0; i < rowCount; ++i)
     {
-    QString segmentID = iterableSegmentIds[segmentIndex];
-    if (currentSegmentID == segmentID)
+    newRow = std::max(0, (newRow + segmentIndexOffset) % rowCount);
```

Comment:
What about adding a comment briefly explaining the relationship of row index and segmentIndexOffset ?